crux-core/src/crux/moberg.clj

```diff
@@ -220,19 +220,20 @@
       (delay {:crux.tx/tx-id (.id m)
               :crux.tx/tx-time (.time m)})))
 
-  (new-tx-log-context [this]
-    (kv/new-snapshot event-log-kv))
-
-  (tx-log [this tx-log-context from-tx-id]
-    (let [i (kv/new-iterator tx-log-context)]
-      (when-let [m (seek-message i ::event-log from-tx-id)]
-        (for [^Message m (->> (repeatedly #(next-message i ::event-log))
-                              (take-while identity)
-                              (cons m))
-              :when (= :txs (get (.headers m) :crux.tx/sub-topic))]
-          {:crux.tx.event/tx-events (.body m)
-           :crux.tx/tx-id (.message-id m)
-           :crux.tx/tx-time (.message-time m)}))))
+  (open-tx-log-iterator [this from-tx-id]
+    (let [snapshot (kv/new-snapshot event-log-kv)
+          i (kv/new-iterator snapshot)]
+      (db/->closeable-tx-log-iterator
+       #(.close ^Closeable snapshot)
+       (lazy-seq
```

Comment:
in hindsight, let's move this lazy-seq call to `->closeable-tx-log-iterator` - it'll be clearer there that we're trying to avoid an NPE than here, and prevent new implementors from forgetting it